### DISCUSSION TITLE
How can I prevent the widget in the middle area from moving to the left or right?

### DISCUSSION THREAD

**Discussion by baoqing1124** [eclipse-theia/theia#13968] [Q&A]
I want to restrict the widget from moving to other areas? How can I do that?

**Comment by msujew** [comment]
Hey @baoqing1124,

The arduino IDE does this by [disabling all drag and drop of widgets](https://github.com/arduino/arduino-ide/blob/2e78e96b75b8c8c3971f6b115eec558ba84d0317/arduino-ide-extension/src/browser/theia/core/application-shell.ts#L104-L115). I believe you can do this more selectively by looking at the event data and only disabling the event when your widget is the target of the event.
